core/server/master/src/main/java/alluxio/master/FaultTolerantAlluxioMasterProcess.java

```diff
@@ -73,16 +73,25 @@ protected FaultTolerantAlluxioMasterProcess(JournalSystem journalSystem,
   public void start() throws Exception {
     mRunning = true;
     mJournalSystem.start();
+
+    startMasters(false);
+    LOG.info("Secondary started");
+
+    if (ServerConfiguration.getBoolean(PropertyKey.MASTER_JOURNAL_CATCHUP_PROTECT_ENABLED)) {
+      mJournalSystem.waitForCatchup();
```

Comment:
If this fails, do we still want to proceed?

Reply:
Yeah, I have a `alluxio.master.ufs.journal.max.catchup.time` for users to set the maximum catchup time they want. If the master hasn't finish catching up, timeout in waiting for catchup, it will
1) continue catching up if it's not selected as the primary
2) continue catching up in the gainPrimacy() if selected as the primary.
So it's actually fine for catchup to fail.

The `waitForCatchup` is the best effort for the master to catch up, but doesn't guarantee catchup can be finished within timeout. Even if it doesn't catchup, in both cases described above, it will not introduce issues.